### PR TITLE
Force DriverEndpoint for odsp

### PR DESCRIPTION
**Pull request by NicholasCouri**
## Description

Addressing [AB3768](https://dev.azure.com/fluidframework/internal/_workitems/edit/3768) by forcing the endpointName on the ODSP run.

### CODE REVIEW

**Comment by anthony-murphy** [comment]
I already have a PR out to fix this the right way: https://github.com/microsoft/FluidFramework/pull/14869

**Comment by NicholasCouri** [comment]
> I already have a PR out to fix this the right way: #14869

Awesome!  I will close this one. BTW. Just looked at your PR and don't we need to change the package.json as well ? I mean, to enforce the endpointName ?

File: packages/test/test-service-load/package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
 		"start": "node ./dist/nodeStressTest.js",
 		"start:frs": "node ./dist/nodeStressTest.js --driver routerlicious --driverEndpoint frs --profile ci_frs",
 		"start:mini": "node ./dist/nodeStressTest.js  --profile mini",
-		"start:odsp": "node ./dist/nodeStressTest.js --driver odsp",
+		"start:odsp": "node ./dist/nodeStressTest.js --driver odsp --driverEndpoint odsp",
```
Comment (vladsud) [inline packages/test/test-service-load/package.json:41]: @NicholasCouri , can you please help me understand how it works? When I search for "driverEndpoint" in our repo, I see things like type DriverEndpoint = RouterliciousEndpoint | OdspEndpoint.
I do see places where we use odsp-df strings in code, but it feels like it's a bit messed up.

I feel like we driverEndpoint should be either frs or odsp.
But we also need to add "ring" sub-item, that can be odsp or odsp-df (and OdspEndpoint should be OdspRing, not end-point).

What do you think? Should we open an item to do so? It's not just about using consistent names, it also makes it much easier to work with telemetry, as today we search for "driver" when searching for odsp data, and for "driverEndpoint" when searching for afr data - very inconsistent and confusing.

**Comment by NicholasCouri** [comment]
@Vlad ***@***.***>   Did you look at @Tony's PR : https://github.com/microsoft/FluidFramework/pull/14869? I agree it is confusing and perhaps @***@***.***> can share some light and background here before we take some action.



________________________________

**Comment by NicholasCouri** [comment]
It all comes down to our test drivesr:
FluidFramework/interfaces.ts at main * microsoft/FluidFramework (github.com)<https://github.com/microsoft/FluidFramework/blob/main/packages/test/test-driver-definitions/src/interfaces.ts>

In the beginning we only had driverType. Then it was expanded to include endpoint, but endpoint has always been optional, and has been haphazardly iterated on.

If I were to improve this, I would keep driver type, which is more about which classes we use. I would make endpoint required, and let test drivers put whatever they want their, so a combination of server for something like frs, and ring for odsp, it could be a combo, like frs-canary (not real, just example). This would at least suffice for current usage. If ring and server need to be separated, it would be a bigger change, but I don't see a distinct need for that at the moment.

-Tony

**Pull request closed without merging**